To-Do-app/src/Components/AddTask.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react'
+import { Grid, TextField, makeStyles } from '@material-ui/core'
+import { ITask } from '../Interface'
+import TodoTaskList from './TodoTaskList'
+import { Button } from '@mui/material'
+import Dialog from '@mui/material/Dialog'
+import DialogActions from '@mui/material/DialogActions'
+import DialogContent from '@mui/material/DialogContent'
+import DialogContentText from '@mui/material/DialogContentText'
+import DialogTitle from '@mui/material/DialogTitle'
+import { useForm } from 'react-hook-form'
+
+const useStyles = makeStyles({
+
+    validText: {
+        fontSize: 13,
+        color: '#ee2b2b'
+    }
+})
+
+function AddTask() {
+
+    const classes = useStyles()
+    const [todoList, setTodoList] = useState<ITask[]>([])
+    const [open, setOpen] = useState(false)
+
+    const handleClickOpen = () => {
+        reset()
+        setOpen(true)
+    }
+    const handleClose = () => {
+        reset()
+        setOpen(false)
+    }
+
+    const { register, handleSubmit, reset, formState: { errors } } = useForm<ITask>()
+    const onSubmit = (data: ITask) => {
+        const id = todoList.length + 1
+        const date = new Date()
+        const newTask = { taskName: data.taskName, description: data.description, id: id, date: date }
+        // todoList.push(newTask)
+        setTodoList([...todoList, newTask])
+        reset()
+        handleClose()
+    }
+
+    return (
+        <>
```

Comment:
why do you need this tag ?

Reply:
It shows an error unreachable code, I tried now it is working without that tag in AddTask.tsx